### ISSUE TITLE
Tokenizer padding is right but bloom padding is left

### ISSUE THREAD

**Issue by cgr71ii** [baoguangsheng/fast-detect-gpt#28]
Hi!

First, thanks for your work :) I took a look at your code because I was trying to reproduce the results for `bigscience/bloom-7b1`. But I observed that in your code, by default, the padding side is right for all models except for the pubmed corpus, which is left (why is the pubmed an exception for padding?):

https://github.com/baoguangsheng/fast-detect-gpt/blob/ad79b4737da4cded191de37b960e154317c7f2f9/scripts/model.py#L59-L62

The thing is that I noticed in https://huggingface.co/bigscience/bloom-7b1/blob/main/tokenizer_config.json that the default padding side is left instead of right. I did a little experiment with the bloom tokenizer and the applied padding is left by default (as specified in the file I linked), but in the code the side is right. Is this a bug in your code or is there an explanation?

Thanks!

**Comment by cgr71ii** [comment]
Also, I noticed that in the tokenized input and logits you remove the first token and last token, respectively. Do I understand correctly that this is because you do not want to take into account the BoS and EoS token for the inputs and inferred tokens, respectively?

https://github.com/baoguangsheng/fast-detect-gpt/blob/ad79b4737da4cded191de37b960e154317c7f2f9/scripts/fast_detect_gpt.py#L101
https://github.com/baoguangsheng/fast-detect-gpt/blob/ad79b4737da4cded191de37b960e154317c7f2f9/scripts/fast_detect_gpt.py#L103

If this is the case, for some reason, the bloom tokenizer is not adding either of these tokens by default, but I do not know if this is a bug of the bloom tokenizer or not, because they have specified these tokens in their tokenizer configuration, but they specified `"special_tokens_map_file": null` although they had a special tokens map (https://huggingface.co/bigscience/bloom-7b1/blob/main/special_tokens_map.json), and this might may be the reason why these tokens are not added, but I'm not if this is a bug or not.

**Comment by baoguangsheng** [comment]
A1: The padding side does not affect the detection results. Padding is only employed during batch processing, whereas in our tests, samples are assessed individually. The special code for Pubmed is a legacy from DetectGPT, and we have decided not to alter it.

A2: The reason for the unique handling of the first and last tokens is that they do not meet the basic condition  for metric computation. Firstly, there is no valid predictive distribution available for the first token (no corresponding logit for the first label). Secondly, there is no valid label for the next token of the last token (no corresponding label for the last logit). Additionally, excluding few tokens from the metric does not impact the results obviously because the statistical nature of the method.

Hope it's helpful.

**Comment by cgr71ii** [comment]
Ohhh, I see! Thanks for the clarification. I agree with all your arguments, except that if you were not assessing samples individually, I think that the evaluation wouldn't be fair/valid for the models that use a different padding scheme (e.g., bloom), because the model wouldn't be used in the way it was trained, likely leading to unknown model behaviour.
